### PR TITLE
Dispatch on make horizontal space

### PR DESCRIPTION
**Pull request by charleskawczynski**
This is another peel off from #1289. These changes were approved in that #1289.

### CODE REVIEW

**Review by simonbyrne — APPROVED** [review]

File: examples/hybrid/driver.jl [hunk 1]
```diff
@@ -205,7 +201,8 @@ walltime = @elapsed sol = OrdinaryDiffEq.solve!(integrator)
 
 if is_distributed # replace sol.u on the root processor with the global sol.u
     if ClimaComms.iamroot(comms_ctx)
-        global_h_space = make_horizontal_space(horizontal_mesh, npoly)
+        global_h_space =
+            make_horizontal_space(horizontal_mesh, npoly, comms_ctx)
```
Comment (charleskawczynski) [inline examples/hybrid/driver.jl:205]: Ahh, I bet this is it!

Comment (charleskawczynski) [reply examples/hybrid/driver.jl]: We construct a singleton space for the global space.

**Pull request merged**